### PR TITLE
AR Scan Eligible

### PR DESCRIPTION
**Pull request by LordJavi**
## Description
Add AR Mapping label to the map

## Motivation and Context
The label could be interesting for map users like EX gym is
Also is added to webhook post in order to do possible notification filters with poracle, whmgr, pokealarm, etc.

:warning: Copy the following img to **static/img/misc/ar.png**
ar

## How Has This Been Tested?
locally

## Screenshots (if appropriate):
Screenshot from 2020-11-21 21-25-48

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by Fabio1988** [comment]
some of my community members would also appreciate that feature?!
Can we update this branch?

File: Sources/RealDeviceMap/Modell/Pokestop.swift [hunk 1]
```diff
@@ -448,9 +455,10 @@ class Pokestop: JSONConvertibleObject, WebHookEvent, Hashable {
                 INSERT INTO pokestop (
                     id, lat, lon, name, url, enabled, lure_expire_timestamp, last_modified_timestamp, quest_type,
                     quest_timestamp, quest_target, quest_conditions, quest_rewards, quest_template, cell_id, lure_id,
-                    pokestop_display, incident_expire_timestamp, grunt_type, sponsor_id, updated, first_seen_timestamp
+                    pokestop_display, incident_expire_timestamp, grunt_type, sponsor_id, updated, first_seen_timestamp,
+                    ar_scan_eligible
                 )
-                VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, UNIX_TIMESTAMP(), UNIX_TIMESTAMP())
+                VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, UNIX_TIMESTAMP(), UNIX_TIMESTAMP())
```
Comment (Fabio1988) [inline Sources/RealDeviceMap/Modell/Pokestop.swift:461]: @LordJavi please shorten this line :)

**Review by Fabio1988 — APPROVED** [review]

File: Sources/RealDeviceMap/Modell/Pokestop.swift [hunk 2]
```diff
@@ -173,11 +179,12 @@ class Pokestop: JSONConvertibleObject, WebHookEvent, Hashable {
...
-            fortData.activeFortModifier.contains(.troyDiskGlacial) ||
-            fortData.activeFortModifier.contains(.troyDiskMossy) ||
-            fortData.activeFortModifier.contains(.troyDiskMagnetic) ||
+        self.arScanEligible = fortData.isArScanEligible
+        let lastModifiedTimestamp = UInt32(fortData.lastModifiedTimestampMs / 1000)
+        if fortData.activeFortModifier.contains(.itemTroyDisk) ||
+            fortData.activeFortModifier.contains(.itemTroyDiskGlacial) ||
+            fortData.activeFortModifier.contains(.itemTroyDiskMossy) ||
+            fortData.activeFortModifier.contains(.itemTroyDiskMagnetic) {
             fortData.activeFortModifier.contains(.troyDiskRainy) {
```
Comment (Fabio1988) [inline Sources/RealDeviceMap/Modell/Pokestop.swift:188]: this would cause an error.... it have to be part of the if

Comment (Fabio1988) [reply Sources/RealDeviceMap/Modell/Pokestop.swift:188]: fixed

File: Sources/RealDeviceMap/Modell/Pokestop.swift [hunk 3]
```diff
@@ -173,11 +179,12 @@ class Pokestop: JSONConvertibleObject, WebHookEvent, Hashable {
...
         }
         self.enabled = fortData.enabled
-        let lastModifiedTimestamp = UInt32(fortData.lastModifiedMs / 1000)
-        if fortData.activeFortModifier.contains(.troyDisk) ||
-            fortData.activeFortModifier.contains(.troyDiskGlacial) ||
-            fortData.activeFortModifier.contains(.troyDiskMossy) ||
-            fortData.activeFortModifier.contains(.troyDiskMagnetic) ||
+        self.arScanEligible = fortData.isArScanEligible
+        let lastModifiedTimestamp = UInt32(fortData.lastModifiedTimestampMs / 1000)
+        if fortData.activeFortModifier.contains(.itemTroyDisk) ||
```
Comment (Fabio1988) [inline Sources/RealDeviceMap/Modell/Pokestop.swift]: ahhh and you renamed troydisk.... this won't work.

File: Sources/RealDeviceMap/Modell/Pokestop.swift [hunk 4]
```diff
@@ -173,7 +179,8 @@ class Pokestop: JSONConvertibleObject, WebHookEvent, Hashable {
             self.sponsorId = UInt16(fortData.sponsor.rawValue)
         }
         self.enabled = fortData.enabled
-        let lastModifiedTimestamp = UInt32(fortData.lastModifiedMs / 1000)
+        self.arScanEligible = fortData.isArScanEligible
+        let lastModifiedTimestamp = UInt32(fortData.lastModifiedTimestampMs / 1000)
```
Comment (Fabio1988) [inline Sources/RealDeviceMap/Modell/Pokestop.swift:183]: also this I think

**Review by 123FLO321 — DISMISSED** [review]

**Review by Fabio1988 — DISMISSED** [review]

**Comment by kamieniarz** [comment]
I've noticed 2 major bugs:
1. Raids are gone with this PR... Just nothing, 0 raids on the map while I should have over 1k atm.
2. Some AR Pokestops are just missing, mostly in city centers. I thought Quest scan will fix it but after night they're still missing

**Comment by Fabio1988** [comment]
@kamieniarz 
1. strange behaviour, I run it since yesterday, all raids are shown in the map. I use RDM frontend
2. I have to look into it... Have to search for AR-Pokestops in game :D

**Comment by samonthenetuk** [comment]
Looks to be working well although that AR image is too small and looks awful

**Comment by kamieniarz** [comment]
I made some more tests after latest commits and Raids are back - thanks for that. Also seems like all AR gyms were detected but AR pokestops are still missing mostly in city center of each city... that's just odd. Apart from that all seems to be fine

**Comment by Fabio1988** [comment]
better icon:
https://media.discordapp.net/attachments/473886293647163412/847855846817792040/AR_stop_overlay.png

**Pull request merged**